src/Symfony/Bundle/TwigBundle/ContainerAwareRuntimeLoader.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Loads Twig extension runtimes via the service container.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class ContainerAwareRuntimeLoader implements \Twig_RuntimeLoaderInterface
+{
+    private $container;
+    private $mapping;
+
+    public function __construct(ContainerInterface $container, array $mapping)
+    {
+        $this->container = $container;
+        $this->mapping = $mapping;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load($class)
+    {
+        if (!isset($this->mapping[$class])) {
+            throw new \LogicException(sprintf('Class "%s" is not configured as a Twig runtime. Add the "twig.runtime" tag to the related service in the container.', $class));
```

Comment:
The other possibility is to inject a logger and log such errors. WDYT?

I don't like catching the exceptions in Twig as we try to avoid that as much as possible (mainly for performance reasons).